some char in template can cause compiled go file can not runable.

template
```
value=\"<?= h(aabasdf\Admin\Document::$asdf) ?>\"/>\n
```

compiled go file:
```
package tpl
import (
"bytes"

)
func FailTest() string {
var _buffer bytes.Buffer
_buffer.WriteString("value=\\"<?= h(aabasdf\Admin\Document::$asdf) ?>\\"/>\")


return _buffer.String()
}

```

compiled failed info:
```
/xxx/tpl/failTest.go:8:32: illegal character U+003F '?'
/xxx/tpl/failTest.go:8:44: illegal character U+005C '\'
/xxx/tpl/failTest.go:8:50: illegal character U+005C '\'
/xxx/tpl/failTest.go:8:61: illegal character U+0024 '$'
/xxx/tpl/failTest.go:8:68: illegal character U+003F '?'
/xxx/tpl/failTest.go:8:70: illegal character U+005C '\'
/xxx/tpl/failTest.go:8:71: illegal character U+005C '\'
/xxx/tpl/failTest.go:8:72: string literal not terminated
/xxx/tpl/failTest.go:11:1: expected ')', found 'return'
/xxx/tpl/failTest.go:12:3: expected '}', found 'EOF'
```

--- a/gorazor/cases/bug34.gohtml
+++ b/gorazor/cases/bug34.gohtml
@@ -1,0 +1,1 @@
+value=\"<?= h(aabasdf\Admin\Document::$asdf) ?>\"/>\n

--- a/gorazor/gogen.go
+++ b/gorazor/gogen.go
@@ -74,13 +74,14 @@ func (self *Compiler) genPart() {
 	for _, p := range self.parts {
 		if p.ptype == CMKP && p.value != "" {
 			// do some escapings
-			p.value = strings.Replace(p.value, `\n`, `\\n`, -1)
-			p.value = strings.Replace(p.value, "\n", `\n`, -1)
-			p.value = strings.Replace(p.value, `"`, `\"`, -1)
-			for strings.HasSuffix(p.value, "\\n") {
-				p.value = p.value[:len(p.value)-2]
+			for strings.HasSuffix(p.value, "\n") {
+				p.value = p.value[:len(p.value)-1]
 			}
-			if p.value != "\\n" && p.value != "" {
+			p.value = fmt.Sprintf("%#v", p.value)
+			p.value = p.value[1 : len(p.value)-1]
+			p.value = strings.Replace(p.value, `\t`, "\t", -1)
+
+			if p.value != `\n` && p.value != "" {
 				res += "_buffer.WriteString(\"" + p.value + "\")\n"
 			}
 		} else if p.ptype == CBLK {

--- a/gorazor/test/bug34.go
+++ b/gorazor/test/bug34.go
@@ -1,0 +1,12 @@
+package cases
+
+import (
+	"bytes"
+)
+
+func Bug34() string {
+	var _buffer bytes.Buffer
+	_buffer.WriteString("value=\\\"<?= h(aabasdf\\Admin\\Document::$asdf) ?>\\\"/>\\n")
+
+	return _buffer.String()
+}

